scripts/apps/desks/directives/DeskeditBasic.ts

```diff
@@ -105,6 +89,14 @@ export function DeskeditBasic(desks, WizardHandler, metadata, config,
                 return deskType === 'authoring' && config.features.noPublishOnAuthoringDesk;
             };
 
+            scope.$watch('desk.edit', (newVal, oldVal) => {
+                if ((newVal.content_expiry || oldVal.content_expiry != null) && !_.isEqual(newVal, scope.desk.orig)) {
```

Comment:
Can you watch entire objects and not specify the exact fields?

Reply:
It' about `desk.edit` or `content_expiry`?

Reply:
That's my bad, I think `desk.edit` is not a field, but a copy of the desk entity. I think saving should be enabled when anything in the object changes, so it doesn't make sense to check one or another field. What about iterating fields dynamically and checking all one by one, or generating a diff with a library I mentioned in another comment and enabling saving if the generated diff is not empty?

Reply:
Yup, I am looking onto that library. The issue with the `content_expiry` is that if it is set to `null` in the DB, on editing the desk it set back to `0`, this field always remains in the diff.

Reply:
I'm not sure I understand. If it's `null` in the database and when you edit the desk it becomes `0` it seems to me that it's correct that it's included in the diff. No?

Reply:
Yes. And because of this even if we don't change anything, on changing the tabs confirmation modal keeps popping up, as content_expiry always there included in the diff.

Reply:
You can create a defaults object and apply it immutably on top of the database state before calculating the diff.

```
const database = {
    a: 1,
    b: 2,
    c: null,
};

const defaults = {
    c: 7,
}

const uiState = {
    a: 1,
    b: 2,
    c: 7,
}

diff = diffFunction(applyDefaults(database), uiState) // should produce an empty diff
```